src/main.go

```diff
@@ -891,6 +921,21 @@ func (t *tester) executeStmt(query string) error {
 	return nil
 }
 
+func (t *tester) executeStmtString(query string) (string, error) {
+	var err error
+	if t.tx, err = t.mdb.Begin(); err != nil {
+		t.rollback()
+		return "", err
+	}
+	var result string
+	err = t.tx.QueryRow(query).Scan(&result)
+	if err != nil {
+		return "", err
+	}
+	t.rollback()
```

Comment:
Why we need to `rollback()` here?

Reply:
This is without the rollback:
```
$ ./mysql-tester --record let
INFO[0000] recording tests: [let]                       
WARN[0000] Create new db&{0 0xc000014088 0 {0 0} [0xc0000f26c0] map[] 0 1 0xc0000381e0 false map[0xc0000f26c0:map[0xc0000f26c0:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x4912a0} 
INFO[0000] enable tidb_enable_analyze_snapshot          
./t/let.test: ok! 0 test cases passed, take time 0.000726951 s
ERRO[0000] transaction is not committed correctly, rollback 
INFO[0000] run test [let] ok                            

Great, All tests passed
```

This is with the rollback:
```
$ ./mysql-tester --record let
INFO[0000] recording tests: [let]                       
WARN[0000] Create new db&{0 0xc00012a080 0 {0 0} [0xc000098000] map[] 0 1 0xc00010e180 false map[0xc000098000:map[0xc000098000:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x4912a0} 
INFO[0000] enable tidb_enable_analyze_snapshot          
./t/let.test: ok! 0 test cases passed, take time 0.000983109 s
INFO[0000] run test [let] ok                            

Great, All tests passed
```

Or do you think this should be a `t.commit()` instead? This is for populating variables based on queries, so I don't think anything should modify data.

Reply:
PTAL @bb7133

Reply:
Is a transaction even needed, can not `t.mdb.Query()` be used directly instead?

Reply:
@mjonss yes looks like you're right.

To get a current TS a transaction might be needed, but I don't think this should be made to explicitly support that.

```
sql> SELECT @@tidb_current_ts;
+-------------------+
| @@tidb_current_ts |
+-------------------+
| 0                 |
+-------------------+
1 row in set (0.0004 sec)

sql> START TRANSACTION; SELECT @@tidb_current_ts; ROLLBACK;
Query OK, 0 rows affected (0.0004 sec)
+--------------------+
| @@tidb_current_ts  |
+--------------------+
| 437707649082195969 |
+--------------------+
1 row in set (0.0002 sec)
Query OK, 0 rows affected (0.0001 sec)

```